src/node_nanomsg.cc

```diff
@@ -205,23 +205,21 @@ NAN_METHOD(Shutdown) {
 
 
 NAN_METHOD(Send) {
-    NanScope();
-
-    int s = args[0]->Uint32Value();
-
-    if(!node::Buffer::HasInstance(args[1])) {
-        return NanThrowError("second Argument must be a Buffer.");
-    }
+  NanScope();
 
-    Local<Object> obj = args[1]->ToObject();
-    char* odata = node::Buffer::Data(obj);
-    size_t odata_len = node::Buffer::Length(obj);
-    int flags = args[2]->Uint32Value();
+  int flags = args[2].integer;
+  std::string *input;
 
-    // Invoke nanomsg function.
-    int ret = nn_send(s, odata, odata_len, flags);
+  if (node::Buffer::HasInstance(args[1]->ToObject())) {
+    v8::Handle<v8::Object> object = args[1]->ToObject();
+    const char *data = node::Buffer::Data(object);
+    input = new std::string(data, node::Buffer::Length(object));
+  } else {
+    utf8 str (args[1]->ToString());
+    input = new std::string(*str);
```

Comment:
Where do we delete/free this string? It seems silly to convert the C string to a std::string then back again later using c_str().